internal/client/fetcher.go

```diff
@@ -149,7 +151,23 @@ func (h HTTPFetcher) ReadIssuer(ctx context.Context, hash []byte) ([]byte, error
 
 // FileFetcher knows how to fetch log artifacts from a filesystem rooted at Root.
 type FileFetcher struct {
-	Root string
+	Root                    string
+	IsEntryBundleCompressed bool
+}
+
+// decompress decompresses gzip data if needed
+func (f FileFetcher) decompress(data []byte, compressed bool) ([]byte, error) {
```

Comment:
This func can be "free" - I think it doesn't need to be a method on FileFetcher.